manager/src/main/java/org/openremote/manager/gateway/GatewayService.java

```diff
@@ -684,7 +684,13 @@ protected void processGatewayConnected(String gatewayClientId, String sessionId)
 
         connector.connected(sessionId, createConnectorMessageConsumer(sessionId), () -> {
             clientEventService.closeSession(sessionId);
-            tunnelInfos.values().removeIf(tunnelInfo -> tunnelInfo.getGatewayId().equals(gatewayId));
+            tunnelInfos.values().removeIf(tunnelInfo -> {
+                if(tunnelInfo.getGatewayId().equals(gatewayId)) {
+                    this.stopTunnel(tunnelInfo);
+                    return true;
+                }
+                return false;
+            });
```

Comment:
This is the wrong place to do this; we want to stop and remove tunnels in the `processGatewayChange` method specifically in the `DELETE` case.